src/main/java/org/terasology/metalrenegades/quests/FetchQuestSystem.java

```diff
@@ -0,0 +1,195 @@
+import org.terasology.dynamicCities.buildings.components.SettlementRefComponent;
+import org.terasology.dynamicCities.construction.events.BuildingEntitySpawnedEvent;
+import org.terasology.dynamicCities.parcels.DynParcel;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.inventory.InventoryManager;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.logic.nameTags.NameTagComponent;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.math.geom.Rect2i;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.metalrenegades.economy.systems.CurrencyManagementSystem;
+import org.terasology.network.ClientComponent;
+import org.terasology.registry.In;
+import org.terasology.rendering.nui.Color;
+import org.terasology.tasks.CollectBlocksTask;
+import org.terasology.tasks.Task;
+import org.terasology.tasks.components.QuestComponent;
+import org.terasology.tasks.components.QuestListComponent;
+import org.terasology.tasks.components.QuestSourceComponent;
+import org.terasology.tasks.events.BeforeQuestEvent;
+import org.terasology.tasks.events.QuestCompleteEvent;
+import org.terasology.tasks.events.StartTaskEvent;
+import org.terasology.tasks.systems.QuestSystem;
+import org.terasology.utilities.Assets;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Manages the fetch meat quest
+ */
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class FetchQuestSystem extends BaseComponentSystem {
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private CurrencyManagementSystem currencyManagementSystem;
+
+    @In
+    private InventoryManager inventoryManager;
+
+    @In
+    private QuestSystem questSystem;
+
+    @In
+    private LocalPlayer localPlayer;
+
+    private EntityRef activeQuestEntity;
+    private Map<String, Integer> amounts = new HashMap<>();
+
+    private final String HOME_TASK_ID = "returnHome";
+    private final String FETCH_QUEST_ID = "FetchQuest";
+    private final String ITEM_ID = "WildAnimals:Meat";
+    private final int REWARD = 50;
+
+    @Override
+    public void postBegin() {
+        activeQuestEntity = EntityRef.NULL;
+    }
+
+    @ReceiveEvent(components = GenericBuildingComponent.class)
+    public void onChurchSpawn(BuildingEntitySpawnedEvent event, EntityRef entityRef) {
+        GenericBuildingComponent genericBuildingComponent = entityRef.getComponent(GenericBuildingComponent.class);
+        if (genericBuildingComponent.name.equals("simplechurch")) {
```

Comment:
It think that it would be good if there was a better way to identify the building component. Maybe we should at least define a few String constants so we can track usages and possible renames.